# start point of line
float64 start_x
float64 start_y

# end point of line
float64 end_x
float64 end_y

# slope of line
float64 k

# intercept of line
float64 b

bool is_reverse
